### PR TITLE
Add `package-manager`option

### LINKED ISSUE #65: Coverage run not respecting yarn.lock
Hi, wanted to start by thanking you for this awesome action, its great 👍🏻

Issue: We use yarn.lock in our codebase to lock dependencies in place, due to which our action for test passed, but this action for coverage failed as it pulled in the latest package hence failing our dependent snapshots.

Can we please make some changes so that this action respects the yarn.lock, OR maybe have configuration in a way where we as a user after checkout, do the regular node setup, and yarn install, and this package only runs coverage. Thoughts, suggestions?

### PR DESCRIPTION
**Pull request by taschetto**
# Summary

This PR introduces a new parameter called `package-manager`, which can be either `npm` (default) or `yarn`. This value will be taken into consideration when installing the project dependencies.

## What this PR does

- adds `package-manager` option

Closes #65.

### CODE REVIEW

File: src/collect/getRawCoverage.ts [hunk 1]
```diff
@@ -4,12 +4,21 @@ import { exec } from '@actions/exec';
...
     join(...(segments as string[]).filter((segment) => segment !== undefined));
 
+const getPackageManagerInstallCommand = (
+    packageManager: PackageManagerType
+): string =>
+    packageManager === 'npm'
+        ? 'npm i'
+        : packageManager === 'yarn'
+        ? 'yarn install'
+        : '';
```
Comment (ArtiomTr) [inline src/collect/getRawCoverage.ts:20]: `npm i` is actually shorthand for the `npm install` command, so this function can be:

```ts
const getPackageManagerInstallCommand = (
    packageManager: PackageManagerType
): string => `${packageManager} install`;
```

File: README.md [hunk 2]
```diff
@@ -96,6 +96,16 @@ with:
     test-script: npm run test:coverage
 ```
 
+## Usage with `yarn`
+
+By default, this action will install your dependencies using `npm`. If you are using `yarn`, you can specify it in the `package-manager` option:
+
+```yml
+with:
+    github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (ArtiomTr) [inline README.md:105]: Here is a small typo 
```diff
- github_token:
+ github-token:
```

**Review by ArtiomTr — CHANGES_REQUESTED** [review]
Almost everything is great - just two small remarks

File: README.md (hunk 2 above)
Comment (taschetto) [reply README.md:105]: Bad merge. Thanks!

File: src/collect/getRawCoverage.ts (hunk 1 above)
Comment (taschetto) [reply src/collect/getRawCoverage.ts:20]: Done!

**Comment by taschetto** [comment]
@ArtiomTr no pressure, but could you release `v2` after this gets merged? :grin:

**Review by ArtiomTr — APPROVED** [review]
Great job 👍

**Comment by ArtiomTr** [comment]
> @ArtiomTr no pressure, but could you release `v2` after this gets merged? 😁

Yes, of course, but I would like to release the `v2.0-rc.1` version, since all the necessary functionality for version 2.0 is ready, but I want to tweak the documentation website a little more. Will be it okay for you? Or you want to see version 2.0 now?

**Comment by taschetto** [comment]
@ArtiomTr I believe I can use the `v2.0-rc.1` for the time being. :)

**Comment by ArtiomTr** [comment]
> @ArtiomTr I believe I can use the `v2.0-rc.1` for the time being. :)

Cool! The [`v2.0-rc.1`](https://github.com/ArtiomTr/jest-coverage-report-action/releases/tag/v2.0-rc.1) version is available now!

**Comment by ArtiomTr** [comment]
@all-contributors please add @taschetto for code, doc

**Pull request merged**